networkx/algorithms/simple_paths.py

```diff
@@ -908,15 +913,16 @@ def iterate(v):
             # we have now discovered the shortest path
             return (finaldist, finalpath)
 
-        wt = _weight_function(G, weight)
         for w in neighs[dir](v):
-            if dir == 0:  # forward
-                minweight = wt(v, w, G.get_edge_data(v, w))
-                vwLength = dists[dir][v] + minweight
-            else:  # back, must remember to change v,w->w,v
-                minweight = wt(w, v, G.get_edge_data(w, v))
-                vwLength = dists[dir][v] + minweight
-
+            # wt(v, w, d) for forward and wt(w, v, d) for back direction
+            cost = (
+                wt(v, w, G.get_edge_data(v, w))
+                if dir == 0
+                else wt(w, v, G.get_edge_data(w, v))
+            )
```

Comment:
Can you rewrite these 5 lines as a 4 line version using the same structure and comment lines as the original code? The style linting makes this (arguably) harder to read than a standard if structure. And that's OK. Python resisted the `inline if` for a long time because it makes most code harder to read (and finally relented because it makes other -- more rare cases -- easier to read). From my perspective, this case is easier to read with the standard structure. Hopefully by leaving the original comments in there too we can reduce the diff and resulting blame footprint of this change.  Let's leave the name `minweight` too for that reason.

The main (and very helpful) addition of this PR is the inclusion of `None` as a valid edge weight to mean "ignore this edge".

Can you add a test of this behavior to make sure it works and, more importantly, to make sure we don't create other changes that break this behavior without knowing it.